src/resolve.ts

```diff
@@ -0,0 +1,47 @@
+import {NonspatialScaleChannel, SCALE_CHANNELS, SPATIAL_SCALE_CHANNELS, SpatialScaleChannel, UNIT_SCALE_CHANNELS} from './channel';
+import * as log from './log';
+import {contains} from './util';
+
+export type ResolveMode = 'independent' | 'shared';
+
+export interface SpatialResolve {
+  scale: ResolveMode;
+  axis?: ResolveMode;
+}
+
+export interface NonspatialResolve {
+  scale: ResolveMode;
+  legend?: ResolveMode;
+}
+
+export type Resolve = SpatialResolve | NonspatialResolve;
+
+export function isSpatialResolve(resolve: Resolve): resolve is SpatialResolve {
```

Comment:
Yeah, I do that in the code. I just need this to be able to access the axis or legend propertly.